src/main/java/org/reaktivity/nukleus/kafka/internal/stream/KafkaClientProduceFactory.java

```diff
@@ -989,6 +1075,9 @@ private void cleanupApplication(
 
         private KafkaProduceClientDecoder decoder;
         private int signaledRequestId;
+        private int recordHeaderOffset;
+        private int recordHeaderLimit;
+        private int headerItemsSize;
```

Comment:
Let's rename `headerItemsSize` to `encodeableRecordHeadersBytes`, and always store it in the `encodeSlot` at `encodePool.slotCapacity() - encodeableRecordHeadersBytes` then no need to also track `recordHeaderOffset` and `recordHeaderLimit`, agreed?

Reply:
`recordHeaderOffset` and `recordHeaderLimit` are for record and headersItemSize is for kafkaHeader size which is stored in `encodePool.slotCapacity() - encodeableRecordHeadersBytes`

Reply:
I just released that we will not need `recordHeaderOffset ` and `recordHeaderLimit ` since we need to make sure to construct complete recordHeader at the beginning for the large payload.

Reply:
I removed  `recordHeaderOffset` and `recordHeaderLimit`.